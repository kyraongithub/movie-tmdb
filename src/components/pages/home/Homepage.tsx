import type React from 'react';
import MoviesSlider from '../../container/MoviesSlider/MoviesSlider';
import useHome from './useHome';

const Homepage: React.FC = () => {
  const {
    movieListPlaying,
    isLoadingPlaying,
    movieListPopular,
    isLoadingPopular,
    movieListTopRated,
    isLoadingTopRated,
    movieListUpcoming,
    isLoadingUpcoming,
  } = useHome();

  return (
    <div>
      <MoviesSlider
        isLoading={isLoadingPlaying}
        movieList={movieListPlaying}
        link="now_playing"
      />
      <MoviesSlider
        isLoading={isLoadingPopular}
        movieList={movieListPopular}
        link="popular"
      />
      <MoviesSlider
        isLoading={isLoadingTopRated}
        movieList={movieListTopRated}
        link="top_rated"
      />
      <MoviesSlider
        isLoading={isLoadingUpcoming}
        movieList={movieListUpcoming}
        link="upcoming"
      />
    </div>
  );
};

export default Homepage;
