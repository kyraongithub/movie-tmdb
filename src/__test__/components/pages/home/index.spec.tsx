import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { MemoryRouter } from 'react-router-dom';
import Homepage from '../../../../components/pages/home';

describe('Homepage', () => {
  it('renders Homepage component', () => {
    const queryClient = new QueryClient();
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <MemoryRouter>
          <Homepage />
        </MemoryRouter>
      </QueryClientProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
