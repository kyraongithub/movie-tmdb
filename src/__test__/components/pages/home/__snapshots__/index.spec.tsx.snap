// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage > renders Homepage component 1`] = `
<div>
  <div>
    <div
      class="_skeleton_8f912d"
      style="width: 200px; height: 300px; border-radius: 0px;"
    />
    <div
      class="_skeleton_8f912d"
      style="width: 200px; height: 300px; border-radius: 0px;"
    />
    <div
      class="_skeleton_8f912d"
      style="width: 200px; height: 300px; border-radius: 0px;"
    />
    <div
      class="_skeleton_8f912d"
      style="width: 200px; height: 300px; border-radius: 0px;"
    />
  </div>
</div>
`;
